Sources/MongoSwift/ClientSession.swift

```diff
@@ -1,35 +1,106 @@
+import Foundation
+import mongoc
+
+internal typealias MutableClientSessionPointer = OpaquePointer
+
 /// Options to use when creating a ClientSession.
-internal struct SessionOptions: Encodable {
-    /// Specifies whether read operations should be causally ordered within the session.
-    private let causalConsistency: Bool?
+public struct ClientSessionOptions {}
+
+/// Private helper for providing a `mongoc_session_opt_t` that is only valid within the body of the provided
+/// closure.
+private func withSessionOpts<T>(wrapping options: ClientSessionOptions?,
+                                _ body: (OpaquePointer) throws -> T) rethrows -> T {
+    // swiftlint:disable:next force_unwrapping
+    var opts = mongoc_session_opts_new()! // always returns a value
+    defer { mongoc_session_opts_destroy(opts) }
+    return try body(opts)
 }
 
-/// :nodoc: A session for ordering sequential operations.
-public class ClientSession: Encodable {
-    /// Initializes a new client session.
-    internal init() {
+/// A MongoDB client session.
```

Comment:
Since this is a public type I think we should expand the documentation, or at least link to relevant documentation like [this](https://docs.mongodb.com/manual/reference/server-sessions/).

Reply:
The docs you linked to deal with server sessions, which the end user of the driver doesn't necessarily need to know about and don't really inform the usage of `ClientSession`. Unfortunately, the mongodb documentation on client sessions is neither detailed nor centralized. The best summary is in the 3.6 patch notes, though they're already outdated since there are no mentions of transactions.

So as it is right now, the client session doesn't actually do anything yet. I think as we add features built on client sessions (e.g. causal consistency, transactions), we can add to the `ClientSession` documentation. I think the [python docs](http://api.mongodb.com/python/current/api/pymongo/client_session.html) have a good page on them documented in that manner.

At any rate, I've added a sentence explaining what their purpose is, which is a little better.